src/Polly.Core/Timeout/OnTimeoutArguments.cs

```diff
@@ -3,7 +3,33 @@ namespace Polly.Timeout;
 /// <summary>
 /// Arguments used by the timeout strategy to notify that a timeout occurred.
 /// </summary>
-/// <param name="Context">The context associated with the execution of a user-provided callback.</param>
-/// <param name="Exception">The original exception that caused the timeout.</param>
-/// <param name="Timeout">The timeout value assigned.</param>
-public record OnTimeoutArguments(ResilienceContext Context, Exception Exception, TimeSpan Timeout);
+public sealed class OnTimeoutArguments
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OnTimeoutArguments"/> class.
+    /// </summary>
+    /// <param name="context">The context associated with the execution of a user-provided callback.</param>
+    /// <param name="exception">The original exception that caused the timeout.</param>
+    /// <param name="timeout">The timeout value assigned.</param>
+    public OnTimeoutArguments(ResilienceContext context, Exception exception, TimeSpan timeout)
+    {
+        Context = context;
+        Exception = exception;
```

Comment:
Should we null guard any of these non-nullable types being passed to these argument types, or is it fine if someone forces them to be created with a null?

Reply:
I would say it's not needed as creation is fully under our control.